src/patches.rs

```diff
@@ -89,25 +95,165 @@ fn collect_pickup_resources<'r>(gc_disc: &structs::GcDisc<'r>, starting_items: &
+{   
+    // Get list of all dependencies needed by custom doors //
+    
+    let mut looking_for = HashSet::<_>::new();
+
+    {
+        let looking_for_door: HashSet<_> = DoorType::iter()
+            .flat_map(|pt| pt.dependencies().into_iter())
+            .collect();
+
+        let looking_for_blast_shield: HashSet<_> = BlastShieldType::iter()
+            .flat_map(|pt| pt.dependencies().into_iter())
+            .collect();
+        
+        for (key, value) in looking_for_door.iter() {
+            looking_for.insert((*key, *value));
+        } 
+        for (key, value) in looking_for_blast_shield.iter() {
+            looking_for.insert((*key, *value));
+        }
+    }
+    
+    // Dependencies read from paks and custom assets will go here //
+    let mut found = HashMap::with_capacity(looking_for.len());
+
+    // Remove extra assets from dependency search since they won't appear     //
+    // in any pak. Instead add them to the output resource pool. These assets //
+    // are provided as external files checked into the repository.            //
+    let extra_assets = crate::custom_assets::extra_assets_doors();
+    for res in extra_assets {
+        looking_for.remove(&(res.file_id, res.fourcc()));
+        assert!(found.insert((res.file_id, res.fourcc()), res.clone()).is_none());
+    }
+
+    // Iterate through all paks and add add any dependencies to the resource pool //
+    for pak_name in pickup_meta::ROOM_INFO.iter().map(|(name, _)| name) { // for all paks
+
+        // get the pak //
+        let file_entry = gc_disc.find_file(pak_name).unwrap();
+        let pak = match *file_entry.file().unwrap() {
+            structs::FstEntryFile::Pak(ref pak) => Cow::Borrowed(pak),
+            structs::FstEntryFile::Unknown(ref reader) => Cow::Owned(reader.clone().read(())),
+            _ => panic!(),
+        };
+
+        // Iterate through all resources in the pak //
+        for res in pak.resources.iter() {
+            let key = (res.file_id, res.fourcc());
+            if looking_for.remove(&key) { // If it's one of our dependencies
+                assert!(found.insert(key, res.into_owned()).is_none()); // collect it
+            }
+        }
+    }
+
+    // Generate custom assets (new door variants) //
+    let mut new_assets = vec![];
+
+    for door_type in DoorType::iter() {
+        if door_type.shield_cmdl().to_u32() >= 0xDEAF0000 { // only if it doesn't exist in-game already
+            new_assets.push(create_custom_door_cmdl(&found, door_type));
+        }
+    }
+
+    // Add the newly generated resources //
+    for res in new_assets {
+        let key = (res.file_id, res.fourcc());
+        if looking_for.remove(&key) {
+            assert!(found.insert(key, res).is_none());
+        }
+    }
+
+    if !looking_for.is_empty()
+    {
+        println!("error - still looking for {:?}", looking_for);
+    }
+
     assert!(looking_for.is_empty());
 
     found
 }
 
+fn create_custom_door_cmdl<'r>(
```

Comment:
This probably belongs in custom_assets.rs along side the functions that make custom CMDLs.